src/System.Collections/src/System/Collections/Generic/LinkedList.cs

```diff
@@ -8,20 +8,40 @@
 
 namespace System.Collections.Generic
 {
+    /// <summary>
+    /// A circular doubly linked list.
+    /// </summary>
+     /// <typeparam name="T">Type of data it holds.</typeparam>
     [DebuggerTypeProxy(typeof(ICollectionDebugView<>))]
     [DebuggerDisplay("Count = {Count}")]
     public class LinkedList<T> : ICollection<T>, System.Collections.ICollection, IReadOnlyCollection<T>
     {
-        // This LinkedList is a doubly-Linked circular list.
+        /// <summary>
+        /// The first node in the doubly linked list.
+        /// </summary>
         internal LinkedListNode<T> head;
```

Comment:
I would be fine with leaving out all of the comments on internal fields and leaving only the public docs. Also, there's still some internal fields here that don't have comments.